main/test/src/util/TestEvaluator.scala

```diff
@@ -29,10 +29,9 @@ class TestEvaluator(
     threads: Option[Int] = Some(1),
     outStream: PrintStream = System.out,
     inStream: InputStream = DummyInputStream,
-    debugEnabled: Boolean = false,
-    extraPathEnd: Seq[String] = Seq.empty
+    debugEnabled: Boolean = false
 )(implicit fullName: sourcecode.FullName, tp: TestPath) {
-  val outPath = TestUtil.getOutPath() / extraPathEnd
```

Comment:
@lefou I do not see any point having this `extraPathEnd`. I added `out` to simulate the `out` we usually have, followed by the test path instead of the cmd path, to avoid the test conflicts.

Reply:
I actually rolled back the `extraPathEnd` since it's actually useful.
About using `module.millSourcePath` instead of `TestUtil.getOutPath()`, it's because we need to have the `outPath` inside the source path like it is in "production".

Reply:
In other places, we copied all required sources into the workspace before the test. That way, the tests are stay reproducible.